pr_reviewer.py

```
@@ -125,51 +131,143 @@ def lambda_handler(event, context):
+        # print(i,openai_review_concatenated[i].path)
+        # print(i,openai_review_concatenated[i].position)
+    # print("individual comment")
+    # review_comments_on_lines(files)
+    # print(f'comment posted on #{pr_number} in repository {repo_name}')
     return {
         'statusCode': 200,
         'body': json.dumps(f'comment posted on #{pr_number} in repository {repo_name}')
     }
 
 
 def openai_review_comments(files, pr_number, repo_name):
-    openai_review_concatenated = ''
+    openai_review_concatenated = []
     if len(files) >= 10:
         num_batches = len(files) // 10
         batch_size = (len(files) + num_batches - 1) // num_batches
         batches = [files[i:i + batch_size] for i in range(0, len(files), batch_size)]
         for batch in batches:
             concatenated_patch = ''.join(file.get('patch', '') for file in batch)
-            openai_review_concatenated = openai_review_concatenated + " \n" + generate_openai(
-                get_context(concatenated_patch))
+            batch_review = generate_openai(get_context(concatenated_patch))
+            openai_review_concatenated.extend(batch_review)
     else:
         patch = get_patch_from_pr(pr_number, repo_name)
         context = get_context(patch)
         openai_review_concatenated = generate_openai(context)
+
+    # Convert the list of review comments to a single object
     return openai_review_concatenated
 
+
+
 def generate_openai(context):
-    # with open('system.txt', 'r') as file:
-    #     system = file.read()
+    openai_response = []  # Initialize an empty string to accumulate response content
 
-    openai_response = ''
+    tools = [
+        {
+            "type": "function",
+            "function": {
+                "name": "post_review_comment_on_line",
+                "description": "Post a review comment on a line in a file",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
```

Error handling seems to be missing in post_review_comment_on_line() method. Without error handling, any failure in the request will not be accounted for and could lead to a silent error. Consider adding a check for the response status code and handle errors appropriately.